modules/auxiliary/gather/censys_search.rb

```diff
@@ -45,18 +45,24 @@ def search(keyword, search_type)
       payload = {
         'query' => keyword
       }
-
-      @cli = Rex::Proto::Http::Client.new('www.censys.io', 443, {}, true)
+      @cli = Rex::Proto::Http::Client.new('search.censys.io', 443, {}, true)
       @cli.connect
 
+    if @searchtype.include?('ipv4')
       response = @cli.request_cgi(
-        'method' => 'post',
-        'uri' => "/api/v1/search/#{search_type}",
+        'method' => 'GET',
+        'uri' => "/api/v2/hosts/search?q=#{keyword}",
+        'headers' => { 'Authorization' => basic_auth_header(@uid, @secret) },
+      )
+      res = @cli.send_recv(response)
+    elsif @searchtype.include?('certificates')
+      response = @cli.request_cgi(
+        'method' => 'GET',
+        'uri' => "/api/v1/view/certificates/#{keyword}",
```

Comment:
So make certificate query an option instead of a search type ?

Reply:
Yes, I think that would work. My only concern would be the output format. For hosts with multiple services, the output without certificates should be:
```
[+] 1.14.76.80 - 80/HTTP,139/NETBIOS,3443/HTTP,3790/HTTP,5985/HTTP,8800/HTTP,47001/HTTP
```
When the user set this option to display certificates and since only some services use certificates, the output needs to be reformatted. Maybe something along these lines:
```
[+] 1.14.76.80 - 80/HTTP,139/NETBIOS,3443/HTTP,3790/HTTP,5985/HTTP,8800/HTTP,47001/HTTP
[*] Certificate for 3443/HTTP: C=US, ST=TX, L=Dallas, O=Rapid7, CN=OtherCA
[*] Certificate for 8800/HTTP: C=US, ST=TX, L=Austin, O=Rapid7, CN=AnotherCA
```

Reply:
I'll get to it Sunday when I get back from week end.
Mind to give me a name or pseudo under which I should add you to the module author ?